sunpy/net/vso/vso.py

```diff
@@ -168,6 +170,26 @@ def iter_errors(response):
         if not hasattr(prov_item, 'record') or not prov_item.record:
             yield prov_item
 
+def check_connection(url):
+    try:
+        if requests.get(url).status_code == 200:
+            return True
+        else:
+            return False
+    except (socket.error, socket.timeout) as e:
+        warnings.warn("Connection failed with error {}. \n Retrying with different url and port.".format(e))
+    
+def fallback_test(api,url,port):
+    if api is None:
+        if url is None or port is None:
+            for i in DEFAULT_URL_PORT:
+                url = i["url"]
+                port = i["port"]
+                bool_ans = check_connection(url)
+                if bool_ans:
+                    api = client.Client(url, transport = WellBehavedHttpTransport())
```

Comment:
can you also add the transport to the dictionary, so we can change it for different endpoints? for instance if we are going to use https we would need a different transport object.